src/main/java/com/matera/cadastrocentral/telephone/TelephoneService.java

```diff
@@ -83,7 +87,7 @@ public Telephone patchTelephonePropertyByTelephoneId(UUID telephoneId, Telephone
             Telephone auxTelephone = optionalTelephone.get();
             // Check if client input exists
             if (telephone.getClientId() != null) {
-                if (clientService.getClientById(telephone.getClientId()).isPresent()) {
+                if (clientService.getClientById(telephone.getClientId().getClientId()).isPresent()) {
```

Comment:
Here, I think you’re just overwriting the client's information in the database with the information you received in the DTO, and that can be bad, since it’s a PATCH, not all of the client's information can come in the DTO, so things can be empty in the database, but I'm not sure about that ... Maybe it is worth doing a test to verify this